instrumentation/apache-httpclient/apache-httpclient-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachehttpclient/v4_0/HttpHeaderSetter.java

```diff
@@ -8,9 +8,8 @@
 import io.opentelemetry.context.propagation.TextMapSetter;
 import org.apache.http.client.methods.HttpUriRequest;
 
-class HttpHeadersInjectAdapter implements TextMapSetter<HttpUriRequest> {
-
-  public static final HttpHeadersInjectAdapter SETTER = new HttpHeadersInjectAdapter();
+enum HttpHeaderSetter implements TextMapSetter<HttpUriRequest> {
+  INSTANCE;
```

Comment:
Nit: we don't really need to have setters/getters be enums, we can just create a new instance and pass to the instrumenter.

Reply:
will send a follow-up PR to change in all (few) existing places so @anuraaga can weigh in

Reply:
#3054